lib/rollup.config.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { DEFAULT_EXTENSIONS } from "@babel/core";
+import babel from "@rollup/plugin-babel";
+import commonjs from "@rollup/plugin-commonjs";
+import eslint from "@rollup/plugin-eslint";
+import json from "@rollup/plugin-json";
+import resolve from "@rollup/plugin-node-resolve";
+import replace from "@rollup/plugin-replace";
+import analyze from "rollup-plugin-analyzer";
+import nodePolyfills from "rollup-plugin-node-polyfills";
+import { terser } from "rollup-plugin-terser";
+import typescript from "rollup-plugin-typescript2";
+import workerLoader from "rollup-plugin-web-worker-loader";
+import pkg from "./package.json";
+
+/**
+ * UMD bundle type.
+ *
+ * @constant
+ * @type {string}
+ * @default
+ */
+const UMD_BUNDLE = "umd";
+
+/**
+ * ESM bundle type.
+ *
+ * @constant
+ * @type {string}
+ * @default
+ */
+const ESM_BUNDLE = "esm";
+
+/**
+ * Browser bundle type.
+ *
+ * @constant
+ * @type {string}
+ * @default
+ */
+const BROWSER_BUNDLE = "iife";
```

Comment:
I don't think a browser bundle is necessary here because we don't expect developers to use this plugin directly in a vanilla JS app.